hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueue.kt

```diff
@@ -54,8 +62,100 @@ class HmppsQueue(
+ * Given we don't want to tie up the listener thread there is no advantage of retrying on this thread so
+ *
+ * ```
+ * hmppsQueue.sendMessage(
+ *       eventType = event.eventType,
+ *       event = event.body,
+ *       retryPolicy = NeverRetryPolicy(),
+ *     )
+ * ```
+ *
+ *
+ * Scenario where thread can hang for longer with more retries
+ * ```
+ * hmppsQueue.sendMessage(
+ *       eventType = event.eventType,
+ *       event = event.body,
+ *       retryPolicy = SimpleRetryPolicy().apply { maxAttempts = 10 }
+ *     )
+ * ```
+ *
+ * Scenario where we want to retry many times but use no more than 10 seconds
+ * ```
+ * hmppsQueue.sendMessage(
+ *       eventType = event.eventType,
+ *       event = event.body,
+ *       retryPolicy = SimpleRetryPolicy().apply { maxAttempts = 20 },
+ *       backoffPolicy = FixedBackOffPolicy().apply { backOffPeriod = 500 },
+ *     )
+ * ```
+ *
+ * Scenario where we need to set more SNS attributes
+ * ```
+ * hmppsQueue.sendMessage(
+ *       eventType = event.eventType,
+ *       event = event.body,
+ *       attributes = mapOf(
+ *         "fruit" to MessageAttributeValue.builder().dataType("String").stringValue("banana").build(),
+ *       ),
+ *     )
+ * ```
+ *
+ * @param eventType The type of the event, which clients listen to, e.g., prisoner.movement.added.
+ * @param event The event data in JSON format as a string.
+ * @param delayInSeconds How long to delay the queueing of the message, if required.
+ * @param noTracing Whether to prevent distributed tracing of this message.  This adds noTracing as a message attribute.
+ * @param attributes A map of additional message attributes. eventType is always added as a message attribute, so only supply this if you need additional attributes.
+ * @param retryPolicy The policy for retrying the send request in case of failure. By default, this will retry 3 times after an error.
+ * @param backOffPolicy The policy for backing off between retries. By default, this will be an exponential policy starting at 1 second with a multiplier of 2
+ * @return A response from the send request containing details such as the message ID.
+ */
+fun HmppsQueue.sendMessage(
+  eventType: String,
+  event: String,
+  delayInSeconds: Int? = null,
+  noTracing: Boolean = false,
+  attributes: Map<String, MessageAttributeValue> = emptyMap(),
+  retryPolicy: RetryPolicy = DEFAULT_RETRY_POLICY,
+  backOffPolicy: BackOffPolicy = DEFAULT_BACKOFF_POLICY,
+): SendMessageResponse {
+  val retryTemplate = RetryTemplate().apply {
+    setRetryPolicy(retryPolicy)
+    setBackOffPolicy(backOffPolicy)
+  }
+  val sendMessageRequestBuilder = SendMessageRequest.builder()
+    .queueUrl(queueUrl)
+    .messageBody(event)
+    .messageAttributes(eventTypeSqsMap(eventType, noTracing) + attributes)
+  delayInSeconds?.let { sendMessageRequestBuilder.delaySeconds(delayInSeconds) }
+  val sendMessageRequest = sendMessageRequestBuilder.build()
```

Comment:
The `delayInSeconds?.let` feels wrong here - you're throwing away the result so a `?.run` feels better.

Though you can maybe drop the builder variable entirely with something like:
```
val sendMessageRequest = SendMessageRequest.builder()
    .queueUrl(queueUrl)
    .messageBody(event)
    .messageAttributes(eventTypeSqsMap(eventType, noTracing) + attributes)
    .apply { delayInSeconds?.also { delaySeconds(it) } }
    .build()
```